Platform/OpenNtfsDxe/NTFS.c

```diff
@@ -0,0 +1,376 @@
+
+  ASSERT (This != NULL);
+
+  Instance = (EFI_FS *) This;
+
+  *Root = (EFI_FILE_PROTOCOL *) Instance->RootIndex->File;
+
+  return EFI_SUCCESS;
+}
+
+/**
+   Installs Simple File System Protocol
+**/
+EFI_STATUS
+EFIAPI
+NTFSStart (
+  IN EFI_DRIVER_BINDING_PROTOCOL *This,
+  IN EFI_HANDLE                  Controller,
+  IN EFI_DEVICE_PATH_PROTOCOL    *RemainingDevicePath
+  )
+{
+  EFI_STATUS   Status;
+  EFI_FS       *Instance;
+
+  Instance = AllocateZeroPool (sizeof (EFI_FS));
+  if (Instance == NULL) {
+    DEBUG ((DEBUG_INFO, "NTFS: Could not allocate a new file system instance\n"));
+    return EFI_OUT_OF_RESOURCES;
+  }
+
+  Instance->FileIoInterface.Revision = EFI_SIMPLE_FILE_SYSTEM_PROTOCOL_REVISION;
+  Instance->FileIoInterface.OpenVolume = OpenVolume;
+
+  Instance->DevicePath = DevicePathFromHandle (Controller);
+  if (Instance->DevicePath == NULL) {
+    DEBUG ((DEBUG_INFO, "NTFS: Could not get Device Path\n"));
+    FreePool (Instance);
+    return EFI_NO_MAPPING;
+  }
+
+  Status = gBS->OpenProtocol (
+    Controller,
+    &gEfiBlockIoProtocolGuid,
+    (VOID **) &Instance->BlockIo,
+    This->DriverBindingHandle,
+    Controller,
+    /*
+     * EFI_OPEN_PROTOCOL_BY_DRIVER would return Access Denied here,
+     * because the disk driver has that protocol already open. So use
+     * EFI_OPEN_PROTOCOL_GET_PROTOCOL (which doesn't require us to close it).
+     */
+    EFI_OPEN_PROTOCOL_GET_PROTOCOL
+    );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_INFO, "NTFS: Could not access BlockIO protocol\n"));
+    FreePool (Instance);
+    return Status;
+  }
+
+  Status = gBS->OpenProtocol (
+    Controller,
+    &gEfiDiskIoProtocolGuid,
+    (VOID **) &Instance->DiskIo,
+    This->DriverBindingHandle,
+    Controller,
+    EFI_OPEN_PROTOCOL_BY_DRIVER
+    );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_INFO, "NTFS: Could not access DiskIO protocol\n"));
+    FreePool (Instance);
+    return Status;
+  }
+
+  Instance->EfiFile.Revision    = EFI_FILE_PROTOCOL_REVISION2;
+  Instance->EfiFile.Open        = FileOpen;
+  Instance->EfiFile.Close       = FileClose;
+  Instance->EfiFile.Delete      = FileDelete;
+  Instance->EfiFile.Read        = FileRead;
+  Instance->EfiFile.Write       = FileWrite;
+  Instance->EfiFile.GetPosition = FileGetPosition;
+  Instance->EfiFile.SetPosition = FileSetPosition;
+  Instance->EfiFile.GetInfo     = FileGetInfo;
+  Instance->EfiFile.SetInfo     = FileSetInfo;
+  Instance->EfiFile.Flush       = FileFlush;
+
+  Status = NtfsMount (Instance);
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_INFO, "NTFS: Could not mount file system.\n"));
+    FreePool (Instance);
+    return Status;
+  }
+
+  Status = gBS->InstallMultipleProtocolInterfaces (
+    &Controller,
+    &gEfiSimpleFileSystemProtocolGuid,
+    &Instance->FileIoInterface,
+    NULL
+    );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_INFO, "NTFS: Could not install simple file system protocol\n"));
+    Status = gBS->CloseProtocol (
```

Comment:
Status is assigned but not used, won't this cause compiler warnings with some toolchains?